### PR TITLE
Add management command for collecting tools

### PR DESCRIPTION
**Pull request by dblenkus**
Includes #419.

### CODE REVIEW

File: resolwe/flow/management/commands/collecttools.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            "Are you sure you want to do this?\n"
+            "\n"
+            "Type 'yes' to continue, or 'no' to cancel: ".format(
+                destination=self.destination_path,
+                action=action,
+            )
+        )
+
+        if input(''.join(message)) != 'yes':
+            raise CommandError("Collecting static files cancelled.")
```
Comment (kostko) [inline resolwe/flow/management/commands/collecttools.py]: Too much copy paste? Update error message.

File: resolwe/flow/executors/prepare.py [hunk 2]
```diff
@@ -40,18 +40,14 @@ def extend_settings(self, data_id, files):
 
     def get_tools(self):
```
Comment (tjanez) [inline resolwe/flow/executors/prepare.py]: Can we take this opportunity to rename this to `get_tools_paths`?

We discussed this in the famous #308.

File: resolwe/flow/executors/prepare.py [hunk 3]
```diff
@@ -40,18 +40,14 @@ def extend_settings(self, data_id, files):
...
         """Get tools paths."""
```
Comment (tjanez) [inline resolwe/flow/executors/prepare.py]: tools -> tools'

File: resolwe/flow/management/commands/collecttools.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+import shutil
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+
+from resolwe.flow.utils import get_apps_tools
+
+
+class Command(BaseCommand):
+    """Copies tools files from different locations to the settings.TOOLS_ROOT."""
```
Comment (tjanez) [inline resolwe/flow/management/commands/collecttools.py]: tools -> tools'

File: resolwe/flow/management/commands/collecttools.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+    help = "Collect tools files in a single location."
```
Comment (tjanez) [inline resolwe/flow/management/commands/collecttools.py]: tools -> tools'

File: resolwe/flow/management/commands/collecttools.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+    def add_arguments(self, parser):
+        """Command arguments."""
+        parser.add_argument(
+            '--noinput', '--no-input', action='store_false', dest='interactive',
+            help="Do NOT prompt the user for input of any kind.",
+        )
+
+        parser.add_argument(
+            '-c', '--clear', action='store_true', dest='clear',
+            help="Clear the existing files using the storage before trying to copy or link the "
```
Comment (tjanez) [inline resolwe/flow/management/commands/collecttools.py]: "Clear the existing files using the storage" is not a clear description. Can you improve it?

File: resolwe/flow/management/commands/collecttools.py [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+                destination=self.destination_path,
+                action=action,
+            )
+        )
+
+        if input(''.join(message)) != 'yes':
+            raise CommandError("Collecting static files cancelled.")
+
+    def clear_dir(self):
+        """Delete content of directoryon the given path."""
```
Comment (tjanez) [inline resolwe/flow/management/commands/collecttools.py]: Change to "Delete contents of the directory on the given path."

File: resolwe/flow/management/commands/collecttools.py [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+        self.stdout.write("Deleting content of '{}'.".format(self.destination_path))
```
Comment (tjanez) [inline resolwe/flow/management/commands/collecttools.py]: content -> contents

File: resolwe/flow/management/commands/collecttools.py [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+                os.remove(filename)
+            elif os.path.isdir(filename):
+                shutil.rmtree(filename)
+
+    def change_path_prefix(self, path, old_prefix, new_prefix, app_name):
+        relative_path = os.path.relpath(path, old_prefix)
+        return os.path.join(new_prefix, app_name, relative_path)
+
+    def collect(self):
+        """Get tools locations and copie them to a single location."""
```
Comment (tjanez) [inline resolwe/flow/management/commands/collecttools.py]: tools -> tools'
copie -> copy

File: resolwe/flow/utils/__init__.py [hunk 10]
```diff
@@ -76,3 +79,27 @@ def set_item(dict_or_model, key, value):
         except ValueError:
             set_item(d, k, val)
         return val
+
+
+def get_apps_tools():
+    """Get application tools and their paths.
```
Comment (tjanez) [inline resolwe/flow/utils/__init__.py]: application -> applications'

File: resolwe/flow/utils/__init__.py [hunk 11]
```diff
@@ -76,3 +79,27 @@ def set_item(dict_or_model, key, value):
...
+
+    Return dict of app names as keys and paths to tools directories
```
Comment (tjanez) [inline resolwe/flow/utils/__init__.py]: Change to:
Return a dict with application names as keys and paths to tools' directories as values. Applications without tools are omitted.

**Comment by tjanez** [comment]
@dblenkus, also add a CHANGELOG entry.

File: resolwe/flow/management/commands/collecttools.py [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+    def add_arguments(self, parser):
+        """Command arguments."""
+        parser.add_argument(
+            '--noinput', '--no-input', action='store_false', dest='interactive',
+            help="Do NOT prompt the user for input of any kind.",
+        )
+
+        parser.add_argument(
+            '-c', '--clear', action='store_true', dest='clear',
+            help="Clear the existing files before copy the new files.",
```
Comment (tjanez) [inline resolwe/flow/management/commands/collecttools.py]: Please, change help text to "Clear existing files before copying new files."

**Pull request merged**